SSA/Projects/FullyHomomorphicEncryption/Basic.lean

```diff
@@ -31,7 +31,7 @@ open Polynomial -- for R[X] notation
 We assume that `q > 1` is a natural number (not necessarily a prime) and `n` is a natural number.
 We will use these to build `ℤ/qℤ[X]/(X^(2^n) + 1)`
 -/
-variable (q t : Nat) [ hqgt1 : Fact (q > 1)] (n : Nat)
```

Comment:
it looks like this assumption `hqgt1` is essentially used everywhere - why do we need to factor it out?